pkg/build/builder/common.go

```diff
@@ -81,3 +84,47 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// runScriptInContainer runs a Bash script in an ephemeral Docker container
+// started off imageID, and returns an error if the script cannot be run or
+// returns a non-zero exit code.
+func runScriptInContainer(client DockerClient, script, imageID string) error {
+	glog.Infof("Running post commit hook with image %s ...", imageID)
+
+	// Read CPU and memory limits from /sys/fs/cgroup.
+	// These limits are used to enforce limits on the container that runs
+	// the post commit hook.
+	r := intFromFileReader{}
+	cpuShares := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.shares")
+	cpuPeriod := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_period_us")
+	cpuQuota := r.read("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us")
+	memory := r.read("/sys/fs/cgroup/memory/memory.limit_in_bytes")
+	if r.err != nil {
+		return r.err
+	}
+
+	return dockerRun(client, docker.CreateContainerOptions{
```

Comment:
personally i'd rather not introduce more dependencies on s2i.  It's a pain to fix bugs in s2i code since it requires a bump to openshift.

Instead I'd prefer to minimize the s2i logic openshift is dependent on.

it also feels weird for docker builds to be dependent on s2i code for something (running an image, in this case).